### PR TITLE
CSHARP-2128: Resync CRUD tests to update insertMany test format to a map

### PR DESCRIPTION
**Pull request by ksadoff**
https://jira.mongodb.org/browse/CSHARP-2128
https://evergreen.mongodb.com/version/5d00156f0ae60609069411ee

Ready for interim review

### CODE REVIEW

**Review by vincentkam — APPROVED** [review]
If this passes without any changes to the test runner, awesome! I'd say it's ready for final review 😸

**Comment by ksadoff** [comment]
Ready for final review

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Comment by rstam** [comment]
When you do merge this change the commit comment. It currently does not mention the CSHARP ticket number.

File: tests/MongoDB.Driver.Tests/Specifications/crud/tests/write/insertMany.json [hunk 1]
```diff
@@ -25,10 +25,10 @@
...
       "outcome": {
         "result": {
-          "insertedIds": [
-            2,
-            3
-          ]
+          "insertedIds": {
+            "0": 2,
+            "1": 3
+          }
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Specifications/crud/tests/write/insertMany.json:31]: I gather the runner already supported this format?

**Review by rstam — APPROVED** [review]

File: tests/MongoDB.Driver.Tests/Specifications/crud/tests/write/insertMany.json (hunk 1 above)
Comment (ksadoff) [reply tests/MongoDB.Driver.Tests/Specifications/crud/tests/write/insertMany.json:31]: I believe so. It looks like `InsertMany(...)` is already implemented to support the map format, and when I ran the unit tests after changing the JSON file everything passed.

**Comment by ksadoff** [comment]
Rebased, merged, and pushed.

**Pull request closed without merging**